src/main/java/use_case/EventFromTaskCreator.java

```diff
@@ -0,0 +1,25 @@
+package main.java.use_case;
+
+import main.java.entity.TodoList;
+import main.java.entity_gateway.CalendarManager;
+import main.java.entity_gateway.TaskReader;
+import main.java.entity_gateway.TodoListManager;
+
+public class EventFromTaskCreator implements EventFromTaskCreatorBoundary {
+
+    private final TodoListManager todoListManager;
+    private final CalendarManager calendarManager;
+
+    public EventFromTaskCreator(TodoListManager todoListManager, CalendarManager calendarManager) {
+        this.todoListManager = todoListManager;
+        this.calendarManager = calendarManager;
+    }
+
+    public boolean createEventFromTask(EventFromTaskModel eventData) {
+        TaskReader tr = todoListManager.getTask(0, eventData.getTaskId());
+        CalendarEventModel eventModel = new CalendarEventData(tr.getName(), eventData.getStartTime(), eventData.getStartTime().plus(tr.getDuration()), eventData.getTags(), eventData.getDates());
```

Comment:
Glad you asked. One thing I was thinking about was changing the direct dependency of Event on Task to Event on some identifier for Task (task's id). This deals with the problem of what to do with the original task.
I don't actually have a specific reason for wanting to do this change, but while looking into another topic I found a comment saying something similar, so I think it's a decent direction to go to. [aforementioned comment](https://softwareengineering.stackexchange.com/questions/384208/in-clean-architecture-arent-entities-another-type-of-boundary#comment845083_384213)
It also might help with the persistence system and simplify the Data Sourcing Layer (from Martin Fowler's 3 layer model) whether we implement a Gateway with ORM (Clean Architecture granted Robert Martin says ORM's don't exist) or the Active Record (Martin Fowler's Domain Model). 

This is a larger design decision, though and I think we should flag it to discuss about in more detail for another PR rather than attempting to handle it here.

Reply:
OK I will add it to `Issues` on GitHub as a remainder.